docs/DeveloperGuide.md

```diff
@@ -200,6 +198,44 @@ From the notes in the sequence diagram above, for every attribute in the instrum
 prompt to get user to provide information for that attribute. This is done through a series of methods in
 the `TextUi` class.
 
+### Loading pre-existing data
+The loading of pre-existing data is mainly handled by the `filemanager` and `model` components. The main method calls `loadFileData(instrumentManager)` 
+in the `Storage` class which implements `InstrumentDecoder#readFile(instrumentManager, fileData)`. This method calls 
+`addSavedInstrumentToList(instrumentManager, textSegment)` for each pre-existing instrument which will add the corresponding 
+instrument in the `InstrumentManager` through calling the `addXYZToList` method in the XYZDecoder classes. In the event
+the instrument is not one of the 4 types of instruments, the `InstrumentDecoder` will throw a new `InvalidInstrumentInFileError`
+and display the corresponding error message.
+
+The figures below represents the sequence diagrams when the user loads a pre-existing crypto:
+
+<img src="images/FileManagerSeqBetweenStorageAndDecoder.png" width="800"/>
+
+<img src="images/FileManagerSequenceDiagram.png" width="1040" height="800"/>
+
+More details about the reference frame for adding the decoded instrument into the `InstrumentManager` is shown below:
+
+<img src="images/FileManagerRefDiagCryptoDecoder.png" width="800"/>
```

Comment:
Regarding the frame can hold on first, I asking Rafael if the entities are suppose be inside the grid or can it be outside

Reply:
Ok Rafael got back to me, the entities can be outside the grid, entities have to be inside the grid only if the obj is being created within the ref frame